src/material-experimental/mdc-chips/chip-row.spec.ts

```diff
@@ -226,6 +235,130 @@ describe('MDC-based Row Chips', () => {
         });
       });
     });
+
+    describe('editable behavior', () => {
+      beforeEach(() => {
+        testComponent.editable = true;
+        fixture.detectChanges();
+      });
+
+      it('should apply the mdc-chip--editable class', () => {
+        expect(chipNativeElement.classList).toContain('mdc-chip--editable');
+      });
+
+      it('should begin editing on double click', () => {
+        chipInstance._dblclick(createMouseEvent('dblclick'));
+        expect(chipNativeElement.classList).toContain('mdc-chip--editing');
+      });
+
+      it('should begin editing on ENTER', () => {
+        chipInstance.focus();
+        const primaryActionElement = chipNativeElement.querySelector('.mdc-chip__primary-action')!;
+        chipInstance._keydown(createKeyboardEvent('keydown', ENTER, 'Enter', primaryActionElement));
+        expect(chipNativeElement.classList).toContain('mdc-chip--editing');
+      });
+    });
+
+    describe('editing behavior', () => {
+      let editInputInstance: MatChipEditInput;
+      let chipContentElement: HTMLElement;
+
+      beforeEach(() => {
+        testComponent.editable = true;
+        fixture.detectChanges();
+        chipInstance._dblclick(createMouseEvent('dblclick'));
+        spyOn(testComponent, 'chipEdit');
+        fixture.detectChanges();
+
+        const editInputDebugElement = fixture.debugElement.query(By.directive(MatChipEditInput))!;
+        editInputInstance = editInputDebugElement.injector.get<MatChipEditInput>(MatChipEditInput);
+
+        chipContentElement =
+          chipNativeElement.querySelector('.mat-chip-row-focusable-text-content') as HTMLElement;
+      });
+
+      function keyDownOnPrimaryAction(keyCode: number, key: string) {
+        const primaryActionElement = chipNativeElement.querySelector('.mdc-chip__primary-action')!;
+        chipInstance._keydown(createKeyboardEvent('keydown', keyCode, key, primaryActionElement));
+      }
+
+      it('should not delete the chip on DELETE or BACKSPACE', () => {
+        spyOn(testComponent, 'chipDestroy');
+        keyDownOnPrimaryAction(DELETE, 'Delete');
+        keyDownOnPrimaryAction(BACKSPACE, 'Backspace');
+        expect(testComponent.chipDestroy).not.toHaveBeenCalled();
+      });
+
+      it('should ignore mousedown events', () => {
+        spyOn(testComponent, 'chipFocus');
+        chipInstance._mousedown(createMouseEvent('mousedown'));
```

Comment:
we also have helpers for mouse events https://github.com/angular/components/blob/master/src/cdk/testing/testbed/fake-events/dispatch-events.ts

Reply:
Switched to using dispatchFakeEvent since this doesn't need a realistic event.